### ISSUE TITLE
Whatsapp shares in sign donate share flow

### ISSUE THREAD

**Issue by rachelhbd** [350org/ak_intl_v3#400]
### Context 
This one builds on #399 so it could make sense for the same person to take it on!

Currently our Whatsapp share open in a new tab and takes users out of the Sign > Share > Donate flow, for example on this (real, live) page: https://act.350.org/sign/stop-rosebank/

The whatsapp share button generates this link, which is not tracked:
`https://api.whatsapp.com/send?text=Sign%20this%20petition%20calling%20on%20the%20Government%20to%20reject%20a%20new%20mega-oil%20field%20in%20the%20North%20Sea:%20https://act.350.org/sign/stop-rosebank/ `

For scoping:
1. Is this still the best practice way to offer whatsapp shares overall? (we know some time has passed since this feature was developed)
2. Campaigners have asked if it's possible to open this in a new tab - but we also know this may not be advisable for accessibility, user experience, etc. What would you recommend?
3. Is it possible to add tracking code to the link, instead of having people share the plain link https://act.350.org/sign/stop-rosebank/?
Looking at https://github.com/350org/action-tools/issues/43 for ref, we'd be looking at one of these scenarios (not sure which it would be for this on-page action, but we'll need to figure this out for #399):

> To have users share a page from a link on the page itself, before they've taken an action, give them a clickable link that includes their AKID, if it is known:
> `https://act.350.org/share/link?type=TYPE&page={{page.name}}&akid={{user.akid}}`
> To share after the user has taken an action, such as on a thanks page, you can include the action ID and omit the page and AKID parameters:
> `https://act.350.org/share/link?type=TYPE&action_id={{action.id}}`

We'd also want to add `&short=1` to any link so it's recoded as a shortlink

So, if we go with the first option above (if this works best for the same-page flow), this would generate a link like `https://act.350.org/share/link?type=ot&page_name=stop-rosebank&akid=.2483206.ZpvPZ4&short=1` which when you put it in the browser gets encoded to `https://act.350.org/s/2233093.4hvnAy `

And then The WA share text would look something like this
`https://api.whatsapp.com/send?text=Sign%20this%20petition%20calling%20on%20the%20Government%20to%20reject%20a%20new%20mega-oil%20field%20in%20the%20North%20Sea:%20https://act.350.org/s/2233093.4hvnAy`

**Comment by rachelhbd** [comment]
@cjmabry I think this one may be ready to scope? But I'm not sure it's a priority at the moment, so I'll bump to Jan - but it would also be possible to look at it earlier

**Comment by cjmabry** [comment]
@sukhada you could also take a look at scoping this one as well.

**Comment by sukhada** [comment]
@rachelhbd @cjmabry This is implemented on [this page](https://act.350.org/sign/sukhada-demo-share-page). A few notes/questions:

1) I haven't implemented sharing in a new tab, but I can do so pretty easily. What do you think re: accessibility/usability, @cjmabry?
2) The way we have to implement this would mean that we'd need to update the way that links are included in the Whatsapp share messages now - on the Stop Rosebank page, it looks like they are manually adding the petition URL to the Whatsapp text (see the "Share on Whatsapp – Message" field [here](https://act.350.org/admin/core/petitionpage/16334/change/)). We'd need to update that message so it just reads: "Sign this petition calling on the Government to reject a new mega-oil field in the North Sea:", and the URL will be added programmatically. Would that be an acceptable approach?

**Comment by rachelhbd** [comment]
I think it would be acceptable! Thank you @sukhada

**Comment by cjmabry** [comment]
@sukhada this looks good - could we make the text of the WhatsApp button solid white, rather than what it is now? Having a bit of contrast issue. Seems to be transparent, maybe?

![image](https://user-images.githubusercontent.com/3048335/216430009-1600f19a-8b9e-4aae-a325-9efec465be62.png)


Will hold on any further design feedback ahead of https://github.com/350org/ak_intl_v3/issues/401

**Comment by sukhada** [comment]
@cjmabry this is updated!

**Comment by cjmabry** [comment]
@rachelhbd we had a miscommunication here and went ahead and did this instead of scoping it. Are you okay with us including this in next release or should we hold?

**Comment by cjmabry** [comment]
Tryn mentioned she was having issues QAing this and we believe it was behavior unique to her phone setup, so I'm going to re-QA this one.

**Comment by rachelhbd** [comment]
Thanks @cjmabry for the flag! I also didn't spot the miscommunication before. 

Just had a quick look, and it appears that if no text is entered into the custom page field "Social Media – Share on Whatsapp – Message"
<img width="740" alt="Screenshot 2023-02-17 at 16 43 47" src="https://user-images.githubusercontent.com/22962348/219713258-24c2f4d6-01c1-4687-92af-39d32071ca0b.png">

The preview image just says "None"
<img width="646" alt="Screenshot 2023-02-17 at 16 39 21" src="https://user-images.githubusercontent.com/22962348/219713504-86e8bcb7-6e10-4c43-a23a-22c3ee8ef6a5.png">

And only the link is passed through
<img width="738" alt="Screenshot 2023-02-17 at 16 36 34" src="https://user-images.githubusercontent.com/22962348/219714191-0cecb9f3-2e8c-401a-88ea-7cbca203ec2d.png">

So where there is no text entered into that custom page field, the default WA message should default to the page title text (there is a snippet for that - would need confirming in the AK support docs)

Where there is text entered into that custom page field, it appears to work great! 
<img width="738" alt="Screenshot 2023-02-17 at 16 36 34" src="https://user-images.githubusercontent.com/22962348/219714326-147e76e4-774b-4877-a390-9b5df1952dfa.png">

So I think maybe hold back this one til the above is fixed? And then I think when you're happy with QA this is good to go into a release! 

Once we're ready to release it would be good to give Julia O a headsup first in case it impacts any tests she is running currently

**Comment by sukhada** [comment]
@rachelhbd just updated this with this fix!

**Comment by rachelhbd** [comment]
Amazing that was fast! Looks good 👍 👍

**Comment by cjmabry** [comment]
@tryntastic did you want to take another look at this with the above fix?

**Comment by cjmabry** [comment]
@sukhada did this ever go out? Not finding it in release QA sheets.

**Comment by sukhada** [comment]
@cjmabry no it didn't, for the same reason that [this ticket](https://github.com/350org/action-tools/issues/49) didn't go out.

**Comment by cjmabry** [comment]
@sukhada copy that! Moving this to May and we can resolve.

**Comment by rachelhbd** [comment]
hey @sukhada just catching up again here! Do you think it's possible for the preview to show a placeholder link eg. "https://act.350.org/s/..." or the plain url of the page, and then the API call to get the shorturl is made as a second step after the user clicks on the share link? If it's not workable perhaps we could ask AK for advice!

**Comment by sukhada** [comment]
@rachelhbd I think this should be working now here: https://act.350.org/sign/sukhada-demo-share-page!

**Comment by rachelhbd** [comment]
Thanks @sukhada! Great, yes it looks to be I think - when I hover over the whatsapp link the url shown is the basic one until the user clicks "share on whatsapp"

**Comment by rachelhbd** [comment]
@juliao762 would you mind pointing me to the report where you figured out that shares were getting registered on landing on the page? When this is released it'd be good to triple check we've solved the whole problem!

**Comment by juliao762** [comment]
That was when we used the wrong template set. 

This is response from Mike from AK when he explained what happened:

You're correct to notice that everyone who takes an action on variation A is automatically getting a share link when hitting the thank you page. The reason for this is due to some custom JS in the thanks.html template for the templateset being used for this test:

https://act.350.org/admin/cms/template/21870/edit

jQuery(document).ready(function($){
var shareURL = '/share/link?type=ot&page_name={{[page.name](http://page.name/)}}&short=1';
$.ajax({
method: 'GET',
url: shareURL,
}).done(function(shareURL) {
$('#page-link-url-general').val(shareURL);
});
});

This code is automatically creating a share link for the user of type "ot" (other) and placing it into the webpage. Thus everyone who hits the thanks page is automatically a sharer. I noticed that this JS is not included in the default [350.org](http://350.org/) - v3 International templateset.

On the other hand, when I took an action on the B variation, there were pop-up questions instead, and I only hit the thanks page with the automatic link if I answer yes to the second question to share. If I answer No to both questions, I end up on the [350.org](http://350.org/) homepage.

Best,

Mike


This is the history of this test, but it didn't record what templatesets were used at the time when this mistake happened: 
https://act.350.org/admin/lab/test/225/history/

**Comment by rachelhbd** [comment]
Thanks @juliao762! Do you have a link to the report for that test which enabled you to originally spot the issue by any chance? 

We're still trying to release this updated feature from the Jan templateset which was used in variation A, but with that particular bug fixed!

**Comment by sukhada** [comment]
@rachelhbd I think we can use this report to see the share details: https://act.350.org/report/shares_stats_page/?page_id=16064, essentially it should no longer count as a share every time we go through the sign share donate process. Only when you click "Share on Whatsapp" should the share be tracked.

**Comment by rachelhbd** [comment]
Ah yes! thanks @sukhada that's exactly it

**Comment by rachelhbd** [comment]
Oh sorry, of course! 

I loaded https://act.350.org/sign/sukhada-demo-share-page and tried signing with rachel.hubbard+sharetesty@350.org and clicking on every share type. My Facebook and twitter shares were tracked. WA is just counted as an "Other" share type, but I saw that the number of "Other" shares went up from 51 to 52 after I clicked to share on WA

I also tried signing with rachel.hubbard+sharetestn@350.org and saying "no" to sharing - this didn't appear to be tracked

So I think this is all good, thanks so much!
